log/sequencer.go

```diff
@@ -401,6 +409,9 @@ func (s Sequencer) IntegrateBatch(ctx context.Context, tree *trillian.Tree, limi
 		}
 
 		// Collate node updates.
+		if err := s.prepareLeaves(sequencedLeaves, merkleTree.Size(), label); err != nil {
```

Comment:
I wonder if putting this before `st.update` line is better (because `timeSource.Now()` will be slightly more ahead, and merge delay more pessimistic/realistic).